mobile/native/src/logger.rs

```diff
@@ -0,0 +1,163 @@
+            .collect::<Vec<String>>()
+            .join(",");
+        let msg = format!("{}: {}", event.metadata().name(), fmt_fields);
+
+        LOG_STREAM_SINK
+            .try_get()
+            .expect("StreamSink from Flutter to be initialised")
+            .add(LogEntry {
+                msg,
+                target: event.metadata().target().to_string(),
+                level: event.metadata().level().to_string(),
+            });
+    }
+}
+
+struct Visitor<'a>(&'a mut BTreeMap<String, String>);
+
+impl<'a> tracing::field::Visit for Visitor<'a> {
+    fn record_f64(&mut self, field: &tracing::field::Field, value: f64) {
+        self.0.insert(field.name().to_string(), value.to_string());
+    }
+
+    fn record_i64(&mut self, field: &tracing::field::Field, value: i64) {
+        self.0.insert(field.name().to_string(), value.to_string());
+    }
+
+    fn record_u64(&mut self, field: &tracing::field::Field, value: u64) {
+        self.0.insert(field.name().to_string(), value.to_string());
+    }
+
+    fn record_bool(&mut self, field: &tracing::field::Field, value: bool) {
+        self.0.insert(field.name().to_string(), value.to_string());
+    }
+
+    fn record_str(&mut self, field: &tracing::field::Field, value: &str) {
+        self.0.insert(field.name().to_string(), value.to_string());
+    }
+
+    fn record_error(
+        &mut self,
+        field: &tracing::field::Field,
+        value: &(dyn std::error::Error + 'static),
+    ) {
+        self.0.insert(field.name().to_string(), value.to_string());
+    }
+
+    fn record_debug(&mut self, field: &tracing::field::Field, value: &dyn std::fmt::Debug) {
+        self.0
+            .insert(field.name().to_string(), format!("{:?}", value));
+    }
+}
+
+// Configure and initalise tracing subsystem
+pub fn init_tracing(level: LevelFilter, json_format: bool) -> Result<()> {
+    if level == LevelFilter::OFF {
+        return Ok(());
+    }
+
+    let is_terminal = atty::is(atty::Stream::Stderr);
+
+    // Parse additional log directives from env variable
+    let filter = match std::env::var_os(RUST_LOG_ENV).map(|s| s.into_string()) {
+        Some(Ok(env)) => {
+            let mut filter = log_base_directives(EnvFilter::new(""), level)?;
+            for directive in env.split(',') {
+                match directive.parse() {
+                    Ok(d) => filter = filter.add_directive(d),
+                    Err(e) => println!("WARN ignoring log directive: `{directive}`: {e}"),
+                };
+            }
+            filter
+        }
+        _ => log_base_directives(EnvFilter::from_env(RUST_LOG_ENV), level)?,
+    };
+
+    let fmt_layer = tracing_subscriber::fmt::layer()
+        .with_writer(std::io::stderr)
+        .with_ansi(is_terminal);
+
+    let fmt_layer = if json_format {
+        fmt_layer.json().with_timer(UtcTime::rfc_3339()).boxed()
+    } else {
+        fmt_layer
+            .with_timer(UtcTime::new(format_description!(
```

Comment:
If this is the only place where we need the `time` crate then I would just switch to `UtcTime::rfc_3339()` here aswell.